saleor/payment/gateways/adyen/webhooks.py

```diff
@@ -410,7 +411,11 @@ def handle_pending(notification: Dict[str, Any], gateway_config: GatewayConfig):
 def handle_refund(notification: Dict[str, Any], _gateway_config: GatewayConfig):
     # https://docs.adyen.com/checkout/refund#refund-notification
     transaction_id = notification.get("pspReference")
-    payment = get_payment(notification.get("merchantReference"), transaction_id)
+    payment = get_payment(
+        notification.get("merchantReference"),
+        transaction_id,
+        check_if_active=False,
```

Comment:
`is_active` means "doesn't contribute to the order/checkout" but that doesn't mean that we shouldn't process the refund state in the system